hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/procedure/TestDropPartitionProcedure.scala

```diff
@@ -0,0 +1,237 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hudi.procedure
+
+import org.apache.hudi.common.model.HoodieReplaceCommitMetadata
+import org.apache.hudi.common.table.HoodieTableMetaClient
+import org.apache.hudi.hadoop.fs.HadoopFSUtils
+import org.junit.jupiter.api.Assertions.{assertEquals, assertTrue}
+
+import scala.collection.JavaConverters._
+
+class TestDropPartitionProcedure extends HoodieSparkProcedureTestBase {
+
+  test("Case1: Test Call drop_partition Procedure For Multiple Partitions: '*' stands for all partitions in any partition level") {
+    withTempDir { tmp =>
+      val tableName = generateTableName
+      val tablePath = tmp.getCanonicalPath + "/" + tableName
+      // create table
+      spark.sql(
+        s"""
+           |create table $tableName (
+           |  id int,
+           |  name string,
+           |  price double,
+           |  ts long,
+           |  year string,
+           |  month string,
+           |  day string
+           |) using hudi
+           |tblproperties (
+           | primaryKey = 'id',
+           | preCombineField = 'ts'
+           |)
+           |partitioned by(year, month, day)
+           |location '$tablePath'
+           |
+     """.stripMargin)
+
+      insertData(tableName)
+
+      spark.sql(s"""call drop_partition(table => '$tableName', partition => 'year=2019/month=08/*')""")
+
+      val metaClient = getTableMetaClient(tablePath)
+      val replaceCommitInstant = metaClient.getActiveTimeline.getWriteTimeline
+        .getCompletedReplaceTimeline.getReverseOrderedInstants.findFirst()
+        .get()
+
+      val partitions = HoodieReplaceCommitMetadata
+        .fromBytes(metaClient.getActiveTimeline.getInstantDetails(replaceCommitInstant).get(), classOf[HoodieReplaceCommitMetadata])
+        .getPartitionToReplaceFileIds
+        .keySet()
+
+      assertEquals(3, partitions.size())
+      assertTrue(partitions.asScala.forall(_.startsWith("year=2019/month=08")))
+
+      // clean
+      spark.sql(s"""call run_clean(table => '$tableName', clean_policy => 'KEEP_LATEST_FILE_VERSIONS', file_versions_retained => 1)""")
+      val result = spark.sql(s"""select * from $tableName""").collect()
+      assertEquals(1, result.length)
+    }
+  }
+
+  test("Case2: Test Call drop_partition Procedure For Multiple Partitions: '*' stands for all partitions in any partition level") {
```

Comment:
What's the difference for case2 and case1?

Reply:
case1 shows '*' can be placed in leaf partition while case2 shows it can be placed in middle partition.

> What's the difference for case2 and case1?

case1 shows '*' can be placed in leaf partition while case2 shows it can be placed in middle partition.

Reply:
case1 shows '*' can be placed in leaf partition while case2 shows it can be placed in middle partition.

Reply:
Can we make it clear is the test remindings.